src/main/java/org/truffleruby/core/range/RangeNodes.java

```diff
@@ -223,7 +224,8 @@ protected RubyObjectRange dup(RubyObjectRange range) {
                     getLanguage().objectRangeShape,
                     range.excludedEnd,
                     range.begin,
-                    range.end);
+                    range.end,
+                    range.frozen);
```

Comment:
`.dup` should actually create an unfrozen copy:
```
irb(main):001:0> RUBY_VERSION
=> "3.0.2"
irb(main):002:0> (1..2).frozen?
=> true
irb(main):003:0> (1..2).dup.frozen?
=> false
```

Could you add specs for that (if there isn't already) and fix the behavior?
For the int and long ranges, we need to create non-frozen variants as well then (or create RubyObjectRange, but that seems not advantageous).